jsonapi/data_structs.go

```diff
@@ -48,6 +48,15 @@ func (c *DataContainer) MarshalJSON() ([]byte, error) {
 	return json.Marshal(c.DataObject)
 }
 
+// CustomLink represents a custom link for return in the document.
+type CustomLink struct {
+	Href string                 `json:"href"`
+	Meta map[string]interface{} `json:"meta,omitempty"`
+}
+
+// CustomLinks contains a map of CustomLink objects as given by an element.
+type CustomLinks map[string]CustomLink
+
 // Links is a general struct for document links and relationship links.
 type Links struct {
```

Comment:
this struct could also contain a `map[string]CustomLink` right? Then we could get the standard strings and additional custom stuff. and keep the reference to this struct in the `Data` struct